### PR TITLE
Add a "Data Quality Legend" to the chart plot and added a "Reset Zoom" button to both chart and node plots

### PR DESCRIPTION
**Pull request by igarousi**
List of changes:

- Defined the `generateDataQualityLegend` function in `charts.js` to generate labels for all data quality levels that are used in the plot legend.
- Switched on the legend plugin and used the `generateDataQualityLegend` function to generate labels for the data quality legend. This is currently used only for the timeseries plots. 
- Add a legend title ("Data Quality") 
- Removed the "RESET ZOOM" from the Plot Actions. 
- Added the "RESET ZOOM" button as an overlay element on both plots. They are positioned in the top-right corner.

### CODE REVIEW

**Comment by igarousi** [comment]
@devincowan Could you please review this? This aims to resolve https://cuahsi.atlassian.net/browse/CAM-498 ticket.

File: frontend/src/stores/charts.js [hunk 1]
```diff
@@ -41,6 +41,17 @@ export const useChartsStore = defineStore('charts', () => {
...
+  const generateDataQualityLegend = () => {
+    return dataQualityOptions.map((option) => ({
+      text: option.label,
+      fillStyle: option.pointBorderColor,
+      strokeStyle: option.pointBorderColor,
+      pointStyle: option.pointStyle,
+      strokeStyle: 'black',
+      lineWidth: 2,
+    }));
+  };
```
Comment (devincowan) [inline frontend/src/stores/charts.js:53]:
```suggestion
  const generateDataQualityLegend = () => {
    return dataQualityOptions.map((option) => ({
      text: option.label,
      fillStyle: 'black',
      strokeStyle: option.pointBorderColor,
      pointStyle: option.pointStyle,
      lineWidth: 2,
    }));
  };
```

File: frontend/src/components/LineChart.vue [hunk 2]
```diff
@@ -156,8 +169,29 @@ const options = {
...
+        display: true,
+        text: 'Data Quality',
+        font: {
+          size: 16,
+          weight: 'bold',
+        },
+        padding: {
+          top: 10,
+        },
+      },
```
Comment (devincowan) [inline frontend/src/components/LineChart.vue:194]: Nice!
Also add to `NodeChart.vue`?

File: frontend/src/components/NodeChart.vue [hunk 3]
```diff
@@ -36,14 +48,14 @@
...
                   <v-icon :icon="mdiMagnifyMinusOutline"></v-icon>
                   Zoom to Exent
-                </v-btn>
-                <v-btn @click="resetData()" color="input" class="ma-1">
+                </v-btn> -->
+                <!-- <v-btn @click="resetData()" color="input" class="ma-1">
                   <v-icon :icon="mdiEraser"></v-icon>
                   Reset Data
-                </v-btn>
+                </v-btn> -->
```
Comment (devincowan) [inline frontend/src/components/NodeChart.vue:58]: remove comment

File: frontend/src/components/LineChart.vue [hunk 4]
```diff
@@ -33,10 +46,10 @@
                 <v-icon :icon="mdiCodeJson"></v-icon>
                 Download JSON
               </v-btn>
-              <v-btn @click="resetZoom()" color="input" class="ma-1">
+              <!-- <v-btn @click="resetZoom()" color="input" class="ma-1">
                 <v-icon :icon="mdiMagnifyMinusOutline"></v-icon>
                 Reset Zoom
-              </v-btn>
+              </v-btn> -->
```
Comment (devincowan) [inline frontend/src/components/LineChart.vue:52]: rm comment

File: frontend/src/components/LineChart.vue [hunk 5]
```diff
@@ -7,7 +7,20 @@
...
+          class="zoom-button"
+          color="#f4f4f4"
+          outlined
+          @click="resetZoom()"
+          style="position: absolute; top: 110px; right: 10px; z-index: 10;"
+          >
+          <v-icon :icon="mdiMagnifyMinusOutline" class="me-2"></v-icon>
+          RESET ZOOM
+        </v-btn> 
+        <!-- Chart -->
```
Comment (devincowan) [inline frontend/src/components/LineChart.vue:23]: I like the intent with the reset zoom button!
However, I think it is a bit cumbersome:
image

I would make it smaller -- ideally just an icon, no text, add the text in tooltip.
https://vuetifyjs.com/en/components/buttons/#icon

File: frontend/src/components/NodeChart.vue [hunk 6]
```diff
@@ -8,6 +8,18 @@
...
+        <v-btn
+          class="zoom-button"
+          color="#f4f4f4"
+          outlined
+          @click="resetZoom()"
+          style="position: absolute; top: 80px; right: 520px; z-index: 10;"
+          >
+          <v-icon :icon="mdiMagnifyMinusOutline" class="me-2"></v-icon>
+          RESET ZOOM
+        </v-btn>
```
Comment (devincowan) [inline frontend/src/components/NodeChart.vue:21]: suggest make this just icon with text on tooltip

File: frontend/src/components/StaticMetadata.vue [hunk 7]
```diff
@@ -8,21 +8,42 @@
...
+              <strong>{{ metadataObject.displayKey }}:</strong> {{ metadataObject.value }}
+            </div>
+            <!-- Tooltip Icon -->
+            <v-tooltip location="top" max-width="200px" style="white-space: normal;">
+              <template #activator="{ props }">
+                <v-icon v-bind="props" :icon="mdiInformationOutline" color="primary" style="cursor: pointer; margin-left: 10px;">
+                </v-icon>
+              </template>
+              <div>{{ metadataObject.definition }}</div>
+            </v-tooltip>
```
Comment (devincowan) [inline frontend/src/components/StaticMetadata.vue:25]: it looks like you have a lot of your metadata updates from https://github.com/CUAHSI/SWOT-Data-Viewer/pull/95 included here in this PR... I assume this is either unintentional on your part, or some sort of bug in GH (I've seen diff bugs in GH before...)

File: frontend/src/stores/hydrologic.js [hunk 8]
```diff
@@ -168,17 +169,21 @@ export const useHydrologicStore = defineStore('hydrologic', () => {
...
       abbreviation: 'y',
       definition: 'Latitude of the node or reach ranging from 90°S to 90°N',
       fileType: 'all',
       default: true,
       short_definition: 'Latitude',
+      swotviz_alias: 'Latitude',
       units: 'decimal degrees',
-      plottable: false
+      plottable: false,
+      significant_figures: 4
```
Comment (devincowan) [inline frontend/src/stores/hydrologic.js:186]: appears lots of your metadata updates from https://github.com/CUAHSI/SWOT-Data-Viewer/pull/95 included here in this PR...

**Review by devincowan — CHANGES_REQUESTED** [review]
this is really great initial stab! thank you @igarousi!
I think with a few changes, this would be ready to go!

File: frontend/src/stores/charts.js (hunk 1 above)
Comment (devincowan) [reply frontend/src/stores/charts.js:53]: this suggestion is so the legend actually matches the points on the plot

File: frontend/src/components/LineChart.vue (hunk 2 above)
Comment (devincowan) [reply frontend/src/components/LineChart.vue:194]: (in other words -- you added the legend to the Time charts but not to the Distance charts)

File: frontend/src/components/StaticMetadata.vue (hunk 7 above)
Comment (devincowan) [reply frontend/src/components/StaticMetadata.vue:25]: image
yep looks like you continued off your `IG-metadata-cleanup` branch

Comment (devincowan) [reply frontend/src/components/StaticMetadata.vue:25]: you probably want to rebase `IG-charts-legend` onto `develop`
You will have merge conflicts that might be difficult to resolve... LMK if you need help

Comment (devincowan) [reply frontend/src/components/StaticMetadata.vue:25]: I would suggest that you do this rebase BEFORE addressing any of the other issues in this PR

Comment (devincowan) [reply frontend/src/components/StaticMetadata.vue:25]: Actually I went ahead and rebased for you (hope that I got all the conflicts right! 🤞 )
consider using this branch instead: https://github.com/CUAHSI/SWOT-Data-Viewer/tree/IG-charts-legend-rebase

File: frontend/src/components/LineChart.vue (hunk 5 above)
Comment (devincowan) [reply frontend/src/components/LineChart.vue:23]: maybe @Castronova will have an opinion about the button size...

File: frontend/src/components/StaticMetadata.vue (hunk 7 above)
Comment (devincowan) [reply frontend/src/components/StaticMetadata.vue:25]: Scratch that, the [rebased branch](https://github.com/CUAHSI/SWOT-Data-Viewer/tree/IG-charts-legend-rebase ) did not have the conflicts resolved correctly. So don't use it! 😄 

[IG-charts-legend-2](https://github.com/CUAHSI/SWOT-Data-Viewer/tree/IG-charts-legend-2) should be more accurate, so I would fast-forward merge your branch to that.

File: frontend/src/components/LineChart.vue (hunk 2 above)
Comment (igarousi) [reply frontend/src/components/LineChart.vue:194]: Great suggestion! Done!

File: frontend/src/components/LineChart.vue (hunk 5 above)
Comment (devincowan) [reply frontend/src/components/LineChart.vue:23]: @igarousi here is an example button:
image

```
<v-tooltip>
        <template #activator="{ props }">
          <v-btn
            v-bind="props"
            color="input"
            @click="resetZoom()"
            style="position: absolute; top: 110px; right: 10px; z-index: 10;"
            :icon="mdiMagnifyMinusOutline"
          >
        </v-btn>
        </template>
        RESET ZOOM
      </v-tooltip> 
```

Comment (devincowan) [reply frontend/src/components/LineChart.vue:23]: one other thing on this reset button:
there is no longer any `resetZoom()` function in this vue component. So you will have to figure out how to handle that I think...

Comment (igarousi) [reply frontend/src/components/LineChart.vue:23]: Yes, I just did. Thanks for pointing this out.

File: frontend/src/components/NodeChart.vue (hunk 6 above)
Comment (igarousi) [reply frontend/src/components/NodeChart.vue:21]: Great suggestion! Done!

**Comment by igarousi** [comment]
Got the following error when running "git push -u origin develop"

```
Enumerating objects: 19, done.
Counting objects: 100% (19/19), done.
Delta compression using up to 20 threads
Compressing objects: 100% (7/7), done.
Writing objects: 100% (7/7), 885 bytes | 885.00 KiB/s, done.
Total 7 (delta 6), reused 0 (delta 0), pack-reused 0 (from 0)
remote: Resolving deltas: 100% (6/6), completed with 6 local objects.
remote: error: GH006: Protected branch update failed for refs/heads/develop.
remote:
remote: - 1 review requesting changes by reviewers with write access.
To https://github.com/CUAHSI/SWOT-Data-Viewer.git
 ! [remote rejected] develop -> develop (protected branch hook declined)
error: failed to push some refs to 'https://github.com/CUAHSI/SWOT-Data-Viewer.git'
```

**Comment by devincowan** [comment]
yes I think there are branch protections in place so that you can't PUSH directly to `develop`
https://github.com/CUAHSI/SWOT-Data-Viewer/settings/branches

File: frontend/src/components/LineChart.vue [hunk 9]
```diff
@@ -6,6 +6,22 @@
...
     >
+      <!-- Add Reset Zoom Icon -->
+      <v-tooltip>
+        <template #activator="{ props }">
+          <v-btn
+            v-bind="props"
+            color="input"
+            size="small"
+            @click="resetZoom()"
+            style="position: absolute; top: 120px; right: 45px; z-index: 10;"
```
Comment (devincowan) [inline frontend/src/components/LineChart.vue:17]: image
on LG screens, the button will float right off the plot

Comment (igarousi) [reply frontend/src/components/LineChart.vue:17]: You're correct. I tried to adjust it for large screens, but now it floats to left on a small screen. I mistakenly pushed these changes to the develop, assuming the branch protections were still in active. However, it seems this PR has already been merged. I will create a new PR to address and resolve this issue.

**Pull request merged**